apps/publisher/themes/default/js/lifecycle/lifecycle-core.js

```diff
@@ -514,10 +520,25 @@ var LifecycleUtils = {};
         var transition;
         for (var index = 0; index < transitions.length; index++) {
             transition = transitions[index];
-            actions.push(transition.event);
+            if (currentState == this.currentState) {
+                if (state.allowedActions && state.allowedActions[transition.event]){
```

Comment:
If I combine these ifs UI will render all available actions of current state as allowed actions, even though none of the available actions are allowed for the current role.
Therefore it is necessary to split above if statements as they are added.
This way it is ensured that user who does not have privilege to perform any of the life-cycle transition actions, will not see actions buttons.